### DISCUSSION TITLE
What’s the best way to add Google Address?

### DISCUSSION THREAD

**Discussion by elfeffe** [filamentphp/filament#1922] [Help] [answered]
I have an input, in my actual project, that renders Google Address Autocomplete. 
Then, when the user clicks on the address, I split the street, number, etc into different inputs. 
How is the best way to do it with Filament?

**Comment by studiowizjo** [comment]
That should do the trick:

```php
TextInput::make('address')
    ->reactive()
    ->afterStateHydrated(function (Closure $set, $state) { //or: afterStateUpdated()
	[$city, $street, $number] = explode(', ', $state); //or whatever need to split the data

        $set('city', $city);
        $set('street', $street);
        $set('number', $number);
    })
```

**Comment by danharrin** [comment, marked as answer]
I've written custom Google autocomplete fields in Filament before. Maybe I need to share them in plugins or add them to the core? :)

**Reply by studiowizjo** [reply]
My vote is for adding to the core. That should come in handy :).

**Reply by elfeffe** [reply]
I like plugins because it is great to have good examples of how to make plugins. So plugins are always good.

**Comment by elfeffe** [comment]
Plug-ins I think
